authors/apps/authentication/tests/base_class.py

```diff
@@ -22,16 +22,22 @@ def setUp(self):
         self.url_facebook = reverse('authentication:facebook-auth')
 
     def register_test_user(self):
-        self.client.post(self.url_register,
-                         data=json.dumps(
-                             valid_register_data),
-                         content_type='application/json')
+        response = self.client.post(self.url_register, data=json.dumps(
+                        valid_register_data), content_type='application/json')
+        return response
+
+    def activated_user(self):
+        return User.objects.create_user(
+                                    username='abc123',
+                                    email='abc@abc.com',
+                                    password='ia83naJS')
 
     def register_and_login_test_user(self):
         self.register_test_user()
-        # force authentication for now, since the login system is not implemented yet
-        test_user = User.objects.get(email=valid_login_data['user']['email'])
-        self.client.force_authenticate(user=test_user)
-        # we'll use this when the token based auth system is ready
-        # self.client.credentials(HTTPS_AUTHORIZATION="Bearer {}".format(
-        #     self.client.post(self.url_login, data=valid_login_data, format='json').data["token"]))
+        activation_link = (mail.outbox[0].body.split('\n')).pop(1)
```

Comment:
I think this part of the code needs a bit more of comments.
It is not clear what you are trying to do.

Reply:
Thanks for the feedback @Karuhanga I will go ahead and add a descriptive doc-string. outbox is a great tool to use when testing outgoing emails from Django's `mail` class. Kindly look at [this](https://docs.djangoproject.com/en/2.1/topics/testing/tools/#email-services) documentation.